File: sdk/c/src/partner.c
#include "hitechcloud/partner.h"
#include <stdlib.h>


/* GET /api/partner - Get partner info */
char *hc_partner_get_partner_info(hc_http_client_t *client) {
    return hc_get(client, "/api/partner", NULL);
}

/* POST /api/partner/apply - Apply to partner program */
char *hc_partner_apply(hc_http_client_t *client) {
    return hc_post(client, "/api/partner/apply", NULL);
}

/* GET /api/partner/tiers - List partner tiers */
char *hc_partner_list_tiers(hc_http_client_t *client) {
    return hc_get(client, "/api/partner/tiers", NULL);
}

/* GET /api/partner/pricing - Get partner pricing */
char *hc_partner_get_pricing(hc_http_client_t *client) {
    return hc_get(client, "/api/partner/pricing", NULL);
}

/* GET /api/partner/customers - List customers */
char *hc_partner_list_customers(hc_http_client_t *client) {
    return hc_get(client, "/api/partner/customers", NULL);
}

/* GET /api/partner/earnings - Get earnings */
char *hc_partner_get_earnings(hc_http_client_t *client) {
    return hc_get(client, "/api/partner/earnings", NULL);
}

/* GET /api/partner/wallet - Get wallet balance */
char *hc_partner_get_wallet(hc_http_client_t *client) {
    return hc_get(client, "/api/partner/wallet", NULL);
}

/* GET /api/partner/payouts - List payouts */
char *hc_partner_list_payouts(hc_http_client_t *client) {
    return hc_get(client, "/api/partner/payouts", NULL);
}

/* POST /api/partner/payouts - Request payout */
char *hc_partner_request_payout(hc_http_client_t *client) {
    return hc_post(client, "/api/partner/payouts", NULL);
}

/* GET /api/partner/leads - List leads */
char *hc_partner_list_leads(hc_http_client_t *client) {
    return hc_get(client, "/api/partner/leads", NULL);
}

/* POST /api/partner/leads - Create lead */
char *hc_partner_create_lead(hc_http_client_t *client) {
    return hc_post(client, "/api/partner/leads", NULL);
}

/* GET /api/partner/referral - Get referral info */
char *hc_partner_get_referral(hc_http_client_t *client) {
    return hc_get(client, "/api/partner/referral", NULL);
}

/* GET /api/partner/rates - Get exchange rates */
char *hc_partner_get_rates(hc_http_client_t *client) {
    return hc_get(client, "/api/partner/rates", NULL);
}

File: sdk/r/R/partner.R
#' Partner: partner program, tiers, pricing, customers, earnings, wallet, payouts, leads, referral, rates
#' @param http HttpClient object
#' @return Resource object
#' @export
PartnerResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "PartnerResource"
  env
}

#' GET /api/partner - Get partner info
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getPartnerInfo <- function(self, ...) {
  self$http$get("/api/partner", list(...))
}

#' POST /api/partner/apply - Apply to partner program
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
apply <- function(self, ...) {
  self$http$post("/api/partner/apply", list(...))
}

#' GET /api/partner/tiers - List partner tiers
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listTiers <- function(self, ...) {
  self$http$get("/api/partner/tiers", list(...))
}

#' GET /api/partner/pricing - Get partner pricing
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getPricing <- function(self, ...) {
  self$http$get("/api/partner/pricing", list(...))
}

#' GET /api/partner/customers - List customers
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listCustomers <- function(self, ...) {
  self$http$get("/api/partner/customers", list(...))
}

#' GET /api/partner/earnings - Get earnings
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getEarnings <- function(self, ...) {
  self$http$get("/api/partner/earnings", list(...))
}

#' GET /api/partner/wallet - Get wallet balance
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getWallet <- function(self, ...) {
  self$http$get("/api/partner/wallet", list(...))
}

#' GET /api/partner/payouts - List payouts
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listPayouts <- function(self, ...) {
  self$http$get("/api/partner/payouts", list(...))
}

#' POST /api/partner/payouts - Request payout
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
requestPayout <- function(self, ...) {
  self$http$post("/api/partner/payouts", list(...))
}

#' GET /api/partner/leads - List leads
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listLeads <- function(self, ...) {
  self$http$get("/api/partner/leads", list(...))
}

#' POST /api/partner/leads - Create lead
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
createLead <- function(self, ...) {
  self$http$post("/api/partner/leads", list(...))
}

#' GET /api/partner/referral - Get referral info
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getReferral <- function(self, ...) {
  self$http$get("/api/partner/referral", list(...))
}

#' GET /api/partner/rates - Get exchange rates
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getRates <- function(self, ...) {
  self$http$get("/api/partner/rates", list(...))
}

